### DISCUSSION TITLE
EloquentBuilder join tables based on models

### DISCUSSION THREAD

**Discussion by wizzymore** [laravel/framework#52650] [Ideas]
I request a new feature where something like this would be possible:

```php
class User extends Model {}

class Post extends Model {}

// posts has user_id

User::query()->leftJoin(Post::class)
```

Why is this useful?
- There are a lot of cases where a left join is faster then 2 separate queries ( like laravel currently does on one to one relationships )
- There are a lot of time where i receive a query from the DB administrator and is simpler to first abstract it to the eloquent query builder and then setup the new relations or modify the existing ones
- For the people that need joins it will make sure that modifying the model name and respectively the table (either one, the other or both) will keep both Eloquent and Eloquent Queries working without introducing bugs. Of course manual references of tables will still break but that's why you would use an alias that will be static regardless of the table name

Currently this can be achieved but without the laravel polished feel:
```php
User::query()->leftJoin((new Post)->getTable());

Or:

class CustomModel extends Model {
    public static function getModel(): string {
        return (new self())->getTable(); // or new static() depending on how you will use the class
}
```

**Comment by wizzymore** [comment]
Another thing i could note, i could accept this not being a future that is implemented. But, is there currently a way i could macro the EloquentBuilder to add something like `->leftJoinModel()` ?

**Comment by macropay-solutions** [comment]
Example of macro in Eloquent Builder:
https://github.com/macropay-solutions/laravel-crud-wizard-free/blob/9cb27b576cf8d1bb462296fcf6cf7b14d18c368c/src/Providers/CrudProvider.php#L23

**Reply by wizzymore** [reply]
Thanks!
